### ISSUE TITLE
Geology - maybe symbolize based on `MAJOR1` and add label based on `ORIG_LABEL`; look at documentation for color choices and fill, but make transparent so it can be seen over hillshade and AP

### ISSUE THREAD

**Issue by joewheaton** [Riverscapes/RiverscapesXML#561]

**Comment by jtgilbert** [comment]
fyi @joewheaton @wally-mac @Riverscapes/riverscape-project-curators. This layer has consistent symbology we should use/copy. It's on page 15 of this doc: https://pubs.usgs.gov/ds/1052/ds1052.pdf

**Comment by wally-mac** [comment]
@leallysmith I'm assigning this to you. Please delegate as needed. Thanks.

**Comment by leallysmith** [comment]
![image](https://user-images.githubusercontent.com/89225401/232056851-52893cc1-942b-4753-a389-010f2595c69d.png)
This is the symbology on page 15, it's for the GENERALIZE field which is GENERALIZED_LITH (is this usable in place of MAJOR1? it summarizes MAJOR1-3). Still tracking down symbologies for Unit name and Age.

**Comment by jtgilbert** [comment]
Yeah I think this looks great @leallysmith. As far as the other fields I don't know if SGMC uses any other symbology so you may have to peruse around and find examples from other geologic maps.

**Comment by leallysmith** [comment]
![image](https://user-images.githubusercontent.com/89225401/232092208-0d028f81-d172-447f-8851-eda3d49dc801.png)
This seemed to be a pretty consistent color scheme across different geologic age maps. Should I modify the BL to show both MIN_AGE and MAX_AGE or just one of them? Also, there's about 200 ages that would need to be symbolized. Is there a faster way to symbolize rather than manually assigning colors and typing in every potential age?

**Comment by joewheaton** [comment]
> 
> This seemed to be a pretty consistent color scheme across different geologic age maps. Should I modify the BL to show both MIN_AGE and MAX_AGE or just one of them? 

Just one is fine. I don't care which. 


> Also, there's about 200 ages that would need to be symbolized. Is there a faster way to symbolize rather than manually assigning colors and typing in every potential age? 

Maybe instead of age, just do epoch to keep it simpler.  The color chart from GSA is good.
>

**Comment by joewheaton** [comment]
Also, make label a little smaller font. And perhaps add rule to not display water label if feasible.

**Comment by Cashe93** [comment]
@jtgilbert @joewheaton Here is the updated geology layer. 

![image](https://github.com/Riverscapes/RiverscapesXML/assets/49880208/0b87af56-b834-4434-9974-13eeedb16789)

**Comment by Cashe93** [comment]
@jtgilbert, I'm having a hard time finding an exhaustive list of the UNIT_NAME attributes. Would you know where this is located?

**Comment by Cashe93** [comment]
@jtgilbert below is the Max Age symbology which uses Epochs as suggested by @joewheaton. Given the tedious nature of this task, I'll wait until you guys approve of this before curating it in Mapbox. 

![image](https://github.com/Riverscapes/RiverscapesXML/assets/49880208/2a3cdf8d-afe8-42e6-b1e6-90aafb7572f6)

**Comment by joewheaton** [comment]
This is looking good. Make sure to add the abbreviations used in labels to legend

**Comment by Cashe93** [comment]
Will do @joewheaton. @jtgilbert  do we have the entire national dataset for this layer? If not, could you point me in the direction of where I can find it?

**Comment by Cashe93** [comment]
Disregard @jtgilbert I found it.

**Comment by Cashe93** [comment]
@jtgilbert and @joewheaton, the UNIT_NAME attribute has over 5,500 unique classifications. I can try to use rule-based symbology to group some things together but wanted to confirm that this is worth the time that it will take.

**Comment by jtgilbert** [comment]
I don't think it is worth the time. The GENERALIZE field or whatever it is that is their default symbology covers basic rock types, and between that and age I think you have the most important info. If someone needs to drill in and see exact unit names, or primary and secondary rock types, etc. they can download and query the data.

**Comment by Cashe93** [comment]
Sounds good!

**Comment by Cashe93** [comment]
@jtgilbert, I'm having trouble figuring out a way to symbolize these in Mapbox (specifically the max ages). The AGE_MAX attribute field contains multiple names per row (each name representing eon, era, period, and epoch)  In QGIS we used rule based symbology to isolate the epoch name which required the use of the LIKE operator. For example, in order to symbolize things for the Late-Jurassic epoch, we used the following expression ```"AGE_MAX" LIKE '%Late_Jurassic%'``` However, I'm having a hard time figuring out how to do this in Mapbox. Do you know an expression that we could use in the .JSON that is equivalent to what we did in QGIS?

**Comment by Cashe93** [comment]
@MattReimer, any luck with the rule-based symbology in MapBox?

**Comment by MattReimer** [comment]
I only got a bit of time to look into it but it seems like even simple pattern matching isn't possible (at least not yet).

For now our only option would be to:

1. Symbolize every possible value
2. bake the pattern we want into the actual layer. 

Sorry if that's disappointing. I'm pretty surprised they still can't support this.
